### PR TITLE
feat: remove all legacy status of entity

### PR DESCRIPTION
**Pull request by rainfd**
**Problem:**

**Solution:**

1. replace all mixin.LegacyStatus with mixin.Status
2. replace service_revision status_message string field with record text field.
3. clean unused code of legacy status.

migration sql
```sql
-- Modify "service_revisions" table

-- Modify "status_message" column to "record" column
ALTER TABLE service_revisions ADD COLUMN "record" text NULL;

UPDATE service_revisions SET record = status_message WHERE status_message IS NOT NULL;

ALTER TABLE service_revisions DROP COLUMN status_message;

-- Modify "status" string column to jsonb column
ALTER TABLE service_revisions ADD COLUMN new_status jsonb;

UPDATE service_revisions
SET new_status = jsonb_build_object(
        'summaryStatus', COALESCE(status, 'Failed'),
        'error', status = 'Failed',
        'transitioning', status = 'Running'
    )::jsonb;

ALTER TABLE service_revisions DROP COLUMN status;
ALTER TABLE service_revisions RENAME COLUMN new_status TO status;
```

**Related Issue:**
https://github.com/seal-io/walrus/issues/1225

### CODE REVIEW

File: pkg/apis/dashboard/extension.go [hunk 1]
```diff
@@ -185,26 +183,43 @@ func getServiceRevisionStatusStats(
...
-				Select(
-					sql.As(sql.Count(servicerevision.FieldStatus), "count"),
-					sql.As(groupBy, servicerevision.FieldCreateTime),
-					servicerevision.FieldStatus).
-				GroupBy(
-					groupBy,
-					servicerevision.FieldStatus)
-		}).
-		Scan(ctx, &counts)
+	// Ent doesn't support group by json field yet.
```
Comment (gitlawr) [inline pkg/apis/dashboard/extension.go:186]: Can you try the following?
```
q.
Select(
	sql.As(sql.Count(servicerevision.FieldStatus), "count"),
	sql.As(groupBy, servicerevision.FieldCreateTime),
	sql.As("status ->> 'summaryStatus'", "summaryStatus")).
GroupBy(
	servicerevision.FieldCreateTime,
	"summaryStatus")
```
Let's avoid using raw sql if possible.

File: pkg/deployer/terraform/jobctrl.go [hunk 2]
```diff
@@ -139,17 +141,17 @@ func (r JobReconciler) syncApplicationRevisionStatus(ctx context.Context, job *b
...
 	if job.Status.Failed > 0 {
 		r.Logger.Info("failed", "application-revision", appRevisionID)
-		revisionStatus = status.ServiceRevisionStatusFailed
+		status.ServiceRevisionStatusReady.False(appRevision, "")
 	}
 
 	// Report to application revision.
-	appRevision.Status = revisionStatus
-	appRevision.StatusMessage = revisionStatusMessage
+	appRevision.Record = logs
```
Comment (gitlawr) [inline pkg/deployer/terraform/jobctrl.go:148]: Should we name the field `logs`?

File: pkg/deployer/terraform/deployer.go [hunk 3]
```diff
@@ -396,7 +401,7 @@ func (d Deployer) createRevision(
 		return nil, err
 	}
 
-	if prevEntity != nil && prevEntity.Status == status.ServiceRevisionStatusRunning {
+	if prevEntity != nil && status.ServiceRevisionStatusReady.IsUnknown(prevEntity) {
```
Comment (alexcodelf) [inline pkg/deployer/terraform/deployer.go:404]: Previously, the if condition is equals to a running status. Here, the state is ready

File: pkg/apis/dashboard/extension.go (hunk 1 above)
Comment (rainfd) [reply pkg/apis/dashboard/extension.go:186]: Not working. error message column "status ->> 'summaryStatus'" does not exist

File: pkg/dao/types/status/status_servicerevision.go [hunk 4]
```diff
@@ -0,0 +1,40 @@
+package status
+
+const (
+	ServiceRevisionStatusReady ConditionType = "Ready"
```
Comment (alexcodelf) [inline pkg/dao/types/status/status_servicerevision.go:4]: Only ready status here is too weak to represent different states of revision. Ready used to represent final status.

File: pkg/deployer/terraform/jobctrl.go (hunk 2 above)
Comment (rainfd) [reply pkg/deployer/terraform/jobctrl.go]: replace with name "record"

File: pkg/dao/types/status/status_servicerevision.go (hunk 4 above)
Comment (rainfd) [reply pkg/dao/types/status/status_servicerevision.go:4]: Currently, service revision only contain 3 status, running, succeed and failed, correspond to the 3 types of the ServiceRevisionStatusReady condition one by one. May be add more conditions or status in the future, but not in this pull request.

File: pkg/deployer/terraform/deployer.go (hunk 3 above)
Comment (rainfd) [reply pkg/deployer/terraform/deployer.go:404]: The unknown ServiceRevisionStatusReady condition is stand for the service revision is created but not finished, it means the "running state" as before. More conditions or status has more precise meaning. It's explained below.

**Review by gitlawr — APPROVED** [review]
LGTM

**Pull request merged**